UnitsNet/QuantityFormatter.cs

```diff
@@ -3,186 +3,222 @@
-        /// <see cref="CultureInfo.CurrentCulture" /> if null.</param>
-        /// <remarks>
-        /// The valid format strings are as follows:
-        /// <list type="bullet">
-        ///     <item>
-        ///         <term>A standard numeric format string.</term>
-        ///         <description>Any of the standard numeric format for <see cref="IQuantity.Value" />, except for "G" or "g", which have a special implementation.
-        ///         "C" or "c", "E" or "e", "F" or "f", "N" or "n", "P" or "p", "R" or "r" are all accepted.
-        ///         See https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#standard-format-specifiers.
-        ///         </description>
-        ///     </item>
-        ///     <item>
-        ///         <term>"G" or "g".</term>
-        ///         <description>The value with 2 significant digits after the radix followed by the unit abbreviation, such as "1.23 m".</description>
-        ///     </item>
-        ///     <item>
-        ///         <term>"A" or "a".</term>
-        ///         <description>The default unit abbreviation for <see cref="IQuantity{TUnitType}.Unit" />, such as "m".</description>
-        ///     </item>
-        ///     <item>
-        ///         <term>"A0", "A1", ..., "An" or "a0", "a1", ..., "an".</term>
-        ///         <description>The n-th unit abbreviation for <see cref="IQuantity{TUnitType}.Unit" />. "a0" is the same as "a".
-        ///         A <see cref="FormatException"/> will be thrown if the requested abbreviation index does not exist.</description>
-        ///     </item>
-        ///     <item>
-        ///         <term>"U" or "u".</term>
-        ///         <description>The enum name of <see cref="IQuantity{TUnitType}.Unit" />, such as "Meter".</description>
-        ///     </item>
-        ///     <item>
-        ///         <term>"Q" or "q".</term>
-        ///         <description>The quantity name, such as "Length".</description>
-        ///     </item>
-        /// </list>
-        /// </remarks>
-        /// <returns>The string representation.</returns>
-        public static string Format<TUnitType>(IQuantity<TUnitType> quantity, string? format, IFormatProvider? formatProvider)
-            where TUnitType : Enum
+        else if (format.Length == 1)
         {
-            formatProvider ??= CultureInfo.CurrentCulture;
-            return FormatUntrimmed(quantity, format, formatProvider).TrimEnd();
+            switch (format[0])
+            {
+                case 'G' or 'g':
+                    return FormatWithValueAndAbbreviation(quantity, format, formatProvider);
+                case 'S' or 's':
+                    return ToStringWithSignificantDigitsAfterRadix(quantity, formatProvider, 0);
+                case 'A' or 'a':
+                    return UnitsNetSetup.Default.UnitAbbreviations.GetDefaultAbbreviation(quantity.Unit, formatProvider);
```

Comment:
Please note another "breaking" change (I _propose_): is that we would not longer from throw when given the 'a' format and the quantity's unit doesn't have any abbreviations(e.g. a `Volume` with the `AuTablespoon` unit)- instead we return the `default abbreviation` which is `string.Empty`.